### ISSUE TITLE
has bad ELF magic

### ISSUE THREAD

**Issue by droidwolf** [facebook/fresco#1747]
> `Fatal Exception: java.lang.UnsatisfiedLinkError: dlopen failed: "/data/app/pkg/lib/arm/libgifimage.so" has bad ELF magic
       at java.lang.Runtime.loadLibrary(Runtime.java:375)
       at java.lang.System.loadLibrary(System.java:988)
       at com.facebook.common.soloader.SoLoaderShim$DefaultHandler.loadLibrary(SoLoaderShim.java:32)
       at com.facebook.common.soloader.SoLoaderShim.loadLibrary(SoLoaderShim.java:56)
       at com.facebook.animated.gif.GifImage.ensure(GifImage.java:46)
       at com.facebook.animated.gif.GifImage.create(GifImage.java:68)
       at com.facebook.animated.gif.GifImage.decode(GifImage.java:75)
       at com.facebook.imagepipeline.animated.factory.AnimatedImageFactoryImpl.decodeGif(AnimatedImageFactoryImpl.java:87)
       at com.facebook.imagepipeline.decoder.DefaultImageDecoder.decodeGif(DefaultImageDecoder.java:142)
       at com.facebook.imagepipeline.decoder.DefaultImageDecoder$1.decode(DefaultImageDecoder.java:65)
       at com.facebook.imagepipeline.decoder.DefaultImageDecoder.decode(DefaultImageDecoder.java:123)
       at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.doDecode(DecodeProducer.java:239)
       at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder.access$200(DecodeProducer.java:111)
       at com.facebook.imagepipeline.producers.DecodeProducer$ProgressiveDecoder$1.run(DecodeProducer.java:144)
       at com.facebook.imagepipeline.producers.JobScheduler.doJob(JobScheduler.java:207)
       at com.facebook.imagepipeline.producers.JobScheduler.access$000(JobScheduler.java:27)
       at com.facebook.imagepipeline.producers.JobScheduler$1.run(JobScheduler.java:78)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at com.facebook.imagepipeline.core.PriorityThreadFactory$1.run(PriorityThreadFactory.java:43)
       at java.lang.Thread.run(Thread.java:818)`

device:vivo x7, vivo x7Plus,oppo R8205,oppo R8207(Operating Systems 5/6)

fresco version:1.0.0

**Comment by oprisnik** [comment]
Do you use APK splits? Does this still happen with newer versions of Fresco?

**Comment by oprisnik** [comment]
I assume the problem has been resolved. If not, please let us know.

**Comment by huanting** [comment]
java.lang.UnsatisfiedLinkError: dlopen failed: could not load library "libfb_jpegturbo.so" needed by "libimagepipeline.so"; caused by "libfb_jpegturbo.so" has bad ELF magic(Java:maxMemory=134217728, totalMemory=11235328, freeMemory=2759208; Native:NativeHeapSize=21311488, NativeHeapAllocatedSize=17565344, NativeHeapFreeSize=31176)
at java.lang.Runtime.loadLibrary(Runtime.java:361)
at java.lang.System.loadLibrary(System.java:525)
at com.facebook.common.soloader.SoLoaderShim$DefaultHandler.loadLibrary(SoLoaderShim.java)
at com.facebook.common.soloader.SoLoaderShim.loadLibrary(SoLoaderShim.java)
at com.facebook.imagepipeline.nativecode.ImagePipelineNativeLoader.load(ImagePipelineNativeLoader.java)
at com.facebook.imagepipeline.memory.NativeMemoryChunk.<clinit>(NativeMemoryChunk.java)
at com.facebook.imagepipeline.memory.NativeMemoryChunkPool.alloc(NativeMemoryChunkPool.java)
at com.facebook.imagepipeline.memory.NativeMemoryChunkPool.alloc(NativeMemoryChunkPool.java)
at com.facebook.imagepipeline.memory.BasePool.get(BasePool.java)
at com.facebook.imagepipeline.memory.NativePooledByteBufferOutputStream.<init>(NativePooledByteBufferOutputStream.java)
at com.facebook.imagepipeline.memory.NativePooledByteBufferFactory.newByteBuffer(NativePooledByteBufferFactory.java)
at com.facebook.imagepipeline.memory.NativePooledByteBufferFactory.newByteBuffer(NativePooledByteBufferFactory.java)
at com.facebook.imagepipeline.cache.BufferedDiskCache.readFromDiskCache(BufferedDiskCache.java)
at com.facebook.imagepipeline.cache.BufferedDiskCache.access$400(BufferedDiskCache.java)
at com.facebook.imagepipeline.cache.BufferedDiskCache$2.call(BufferedDiskCache.java)
at com.facebook.imagepipeline.cache.BufferedDiskCache$2.call(BufferedDiskCache.java)
at bolts.Task$4.run(Task.java)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
at java.lang.Thread.run(Thread.java:841)

vivo X3L   Android system version: 4.3
fresco version:  1.2.0
Operation:  happen only once  in many devices

**Comment by lambdapioneer** [comment]
Hi @huanting, does this reproduce on those devices when you compile and run the showcase app from our repository?

If not, can you post the `build.gradle` file of your app?

**Comment by huanting** [comment]
I can't test the showcase app on that device. This bug is reported from the thirdparty's SDk, named umeng.
It only report two times.
below is buid.gradle file.

apply plugin: 'com.android.library'

def config = rootProject.ext.configuration;
def libs = rootProject.ext.libraries;
def test = rootProject.ext.testingLibraries

//task copyLayout(type: Copy) {
// from 'rooban/src/main/res/dialog/messagedialog/.'
// into 'rooban/src/main/res/layout'
//
//}

android {
compileSdkVersion config.compile_sdk
buildToolsVersion config.build_tools

defaultConfig {
    minSdkVersion config.min_sdk
    targetSdkVersion config.target_sdk
    versionCode config.version_code
    versionName config.version_name
}

buildTypes {
    release {
        minifyEnabled false
        proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
    }
}

sourceSets {
    main {
        if (useLibs().equals("INLIBS")) {
            jniLibs.srcDirs = ["libs", "inLibs"]
        } else if (useLibs().equals("OUTLIBS")) {
            jniLibs.srcDirs = ["libs", "outLibs"]
        } else {
            jniLibs.srcDirs = ["libs"]
        }
    }
}
}

dependencies {
compile fileTree(dir: 'libs', include: ['.jar'])
if (useLibs().equals("INLIBS")){
compile fileTree(dir: 'inLibs', include: ['.jar'])
}else if(useLibs().equals("OUTLIBS")){
compile fileTree(dir: 'outLibs', include: ['*.jar'])
}
testCompile "junit:junit:${test.junit}"

compile "com.android.support:recyclerview-v7:${libs.recycler_view}"
compile "com.facebook.fresco:fresco:${libs.fresco}"
compile "com.android.support:percent:${libs.support_version}"
compile "com.android.support:support-v4:${libs.support_version}"
compile "com.mcxiaoke.volley:library:${libs.volley}"
compile "com.google.code.gson:gson:${libs.gson}"
compile "cat.ereza:customactivityoncrash:${libs.customactivityoncrash}"
compile "com.umeng.analytics:analytics:${libs.umeng_analytics}"
compile "com.github.bmelnychuk:atv:${libs.android_tree_view}"
compile "com.github.johnkil.print:print:${libs.johnkil_print}"
compile "com.android.support:multidex:${libs.multidex}"
// compile "com.github.zcweng:switch-button:${libs.switch_button}@aar"

//bugly
compile 'com.tencent.bugly:crashreport:latest.release'
compile 'com.tencent.bugly:nativecrashreport:latest.release'
compile "com.hannesdorfmann.mosby3:viewstate:${libs.mosby}"
compile "com.hannesdorfmann.mosby3:mvp-nullobject-presenter:${libs.mosby}"
}

// if you want use inLibs or outLibs, please config property PROJECT_LIBS_IN_OUT in gradle.properties
// PROJECT_LIBS_IN_OUT value can be INLIBS or OUTLIBS
String useLibs() {
String useLib;
try {
useLib = "${PROJECT_LIBS_IN_OUT}";
} catch (e) {
useLib = "";
}
return useLib;
}

ext {
** libraries = [
recycler_view : "25.1.0",
support_version : "25.1.0",
multidex : "1.0.1",
fresco : "1.2.0",
volley : "1.0.19",
gson : "2.2.4",
customactivityoncrash : "1.2.0",
umeng_analytics : "6.0.9",
johnkil_print : "1.3.1",
mosby : "3.0.4",
android_tree_view : "1.2.+",
annotations : "13.0",
switch_button : "0.0.3"
]
}

**Comment by erikandre** [comment]
The upcoming release of Fresco switches from using SoLoaderShim to SoLoader.
Hopefully this will resolve this issue. If not, please file a new issue or comment here so that this one can be reopened. Thanks!

**Comment by xieyilong** [comment]
Fatal Exception: java.lang.UnsatisfiedLinkError: dlopen failed: "/data/app/com.sten.vte-2/lib/arm/libimagepipeline.so" has bad ELF magic
       at java.lang.Runtime.loadLibrary0 + 994(Runtime.java:994)
       at java.lang.System.loadLibrary + 1533(System.java:1533)
       at com.facebook.common.soloader.SoLoaderShim$DefaultHandler.loadLibrary + 32(SoLoaderShim.java:32)

This problem happened again in my app, below is my build.gralde config 

 ndk {
            abiFilters "armeabi-v7a", "x86"/*,"arm64-v8a","x86_64"*/
        }
 implementation 'com.facebook.fresco:imagepipeline-okhttp3:0.12.0+'
    implementation 'com.facebook.fresco:fresco:0.12.0'

**Comment by xieyilong** [comment]
Fatal Exception: java.lang.UnsatisfiedLinkError: dlopen failed: "/lib/arm/libimagepipeline.so" has bad ELF magic
at java.lang.Runtime.loadLibrary0 + 994(Runtime.java:994)
at java.lang.System.loadLibrary + 1533(System.java:1533)
at com.facebook.common.soloader.SoLoaderShim$DefaultHandler.loadLibrary + 32(SoLoaderShim.java:32)

This problem happened again in my app, below is my build.gralde config

ndk {
abiFilters "armeabi-v7a", "x86"/,"arm64-v8a","x86_64"/
}
implementation 'com.facebook.fresco:imagepipeline-okhttp3:0.12.0+'
implementation 'com.facebook.fresco:fresco:0.12.0'
